src/coreclr/vm/interpexec.cpp

```diff
@@ -972,6 +973,35 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                 ip += 5;
                 break;
             }
+            case INTOP_CALLVIRT:
+            {
+                returnOffset = ip[1];
+                callArgsOffset = ip[2];
+                methodSlot = ip[3];
+
+                MethodDesc *pMD = (MethodDesc*)pMethod->pDataItems[methodSlot];
+
+                OBJECTREF thisArg = LOCAL_VAR(callArgsOffset, OBJECTREF);
+                NULL_CHECK(thisArg);
+
+                // Interpreter-TODO
+                // This needs to be optimized, not operating at MethodDesc level, rather with ftnptr
+                // slots containing the interpreter IR pointer
+                pMD = pMD->GetMethodDescOfVirtualizedCode(&thisArg, pMD->GetMethodTable());
```

Comment:
I think that the `thisArg` needs to be GC protected.

Reply:
Yes, I didn't take GC into account yet. In this case we should just pass the address pointing to the interpreter stack, location which is reported via the GC info.